tests/std/tests/P0896R4_views_iota_death/test.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CONTAINER_DEBUG_LEVEL 1
+
+#include <cassert>
+#include <cstddef>
+#include <ranges>
+#include <vector>
+
+#include <test_death.hpp>
+using namespace std;
+
+template <class I>
+void test_misordered_start_bound_int() {
+    ranges::iota_view{I{42}, I{1}};
+}
+
+template <class UI>
+void test_misordered_start_bound_uint_value_init() {
+    ranges::iota_view<UI, UI>{UI{42}};
+}
+
+template <class T>
+void test_misordered_start_bound_ptr() {
+    T arr[1]{};
+    ranges::iota_view{arr + 1, arr + 0};
+}
+
+int main(int argc, char* argv[]) {
+    std_testing::death_test_executive exec;
+
+#ifdef _DEBUG
+    struct S {};
+
+    exec.add_death_tests({
+        test_misordered_start_bound_int<signed char>, test_misordered_start_bound_int<unsigned char>,
+            test_misordered_start_bound_int<short>, test_misordered_start_bound_int<unsigned short>,
+            test_misordered_start_bound_int<int>, test_misordered_start_bound_int<unsigned int>,
+            test_misordered_start_bound_int<long>, test_misordered_start_bound_int<unsigned long>,
+            test_misordered_start_bound_int<long long>, test_misordered_start_bound_int<unsigned long long>,
+
+            test_misordered_start_bound_int<char>,
+#if __cpp_char8_t

```

Comment:
This will probably emit warnings about `__cpp_char8_t` being undefined; prefer `#ifdef __cpp_char8_t`.

Reply:
But the tests are passing. Do we define `__cpp_char8_t` to `0` or something for `/std:c++20 /Zc:char8_t-`? Or do we not have coverage of `/Zc:char8_t-` in the test matrix?

Reply:
Oh, I should use `#ifdef`. But it's guaranteed that when the macro is not defined, `0` is used instead ([[cpp.cond]/11](https://eel.is/c++draft/cpp.cond#11)).

Reply:
`/W4` includes [C4668](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4668?view=msvc-170) which should be complaining about `#if __cpp_char8_t` if it's not defined. :shrug: